iroha/src/torii.rs

```diff
@@ -1,192 +1,114 @@
+            world_state_view: Arc::new(Mutex::new(WorldStateView::new())),
         }
     }
 
     pub async fn start(&mut self) {
-        let listener = TcpListener::bind(&self.url).expect("could not start server");
-        let world = Arc::clone(&self.world);
-        thread::spawn(move || {
-            world.init();
+        let url = &self.url.clone();
+        let rx = Arc::clone(&self.blocks_receiver);
+        let world_state_view = Arc::clone(&self.world_state_view);
+        thread::spawn(|| {
+            executor::block_on(run_the_world(rx, world_state_view));
         });
-        for connection in listener.incoming() {
-            match connection {
-                Ok(mut stream) => {
-                    stream
-                        .set_read_timeout(Some(Duration::new(2, 0)))
-                        .expect("Failed to set read timeout");
-                    stream
-                        .set_write_timeout(Some(Duration::new(2, 0)))
-                        .expect("Failed to set read timeout");
-                    let mut buffer = [0; 512];
-                    let _read_size = stream.read(&mut buffer).expect("Request read failed.");
-                    if buffer.starts_with(COMMAND_REQUEST_HEADER) {
-                        self.receive_command(&buffer[COMMAND_REQUEST_HEADER.len()..]);
-                        stream.write_all(OK).expect("Failed to write a response.");
-                        self.blockchain
-                            .accept(
-                                self.consensus
-                                    .sign(self.queue.pop_pending_transactions())
-                                    .await
-                                    .expect("Failed to sign transactions."),
-                            )
-                            .await
-                            .expect("Failed to accept transactions into blockchain.");
-                        self.last_round_time = Instant::now();
-                    } else if buffer.starts_with(QUERY_REQUEST_HEADER) {
-                        match self.receive_query(&buffer[QUERY_REQUEST_HEADER.len()..]) {
-                            Ok(result) => {
-                                let mut response = OK.to_vec();
-                                let result = &result;
-                                response.append(&mut result.into());
-                                stream
-                                    .write_all(&response)
-                                    .expect("Failed to write a response.");
-                            }
-                            Err(e) => {
-                                eprintln!("{}", e);
-                                stream
-                                    .write_all(INTERNAL_ERROR)
-                                    .expect("Failed to write a response.");
-                            }
+        let world_state_view = Arc::clone(&self.world_state_view);
+        Network::listen(url, |mut stream| {
```

Comment:
`start` function is `async` but here `Network::listen` and `Network::handle_message` are blocking. Also I haven't seen any other `await` statements in this function. Maybe this is because it's a draft and you will change it later, but just in case.

Reply:
Oh okay, I see you are using the previous version of `networking` where it's difficult to use `async` listen.

Reply:
fixed